### PR TITLE
chore: enforce node 18

### PR DESCRIPTION
**Pull request by vscaiceanu-1a**
## Proposed change

LTS ended for node 16 on [11.09.23](https://endoflife.date/nodejs).

### CODE REVIEW

File: packages/@ama-sdk/schematics/package.json [hunk 1]
```diff
@@ -99,8 +99,8 @@
   },
   "openApiSupportedVersion": "~6.6.0",
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=4",
+    "node": ">=18.0.0",
+    "npm": "please-use-yarn",
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/package.json:103]: nop, only in the private package.json

File: packages/@ama-sdk/core/package.json [hunk 2]
```diff
@@ -122,8 +122,8 @@
     "zone.js": "~0.13.1"
   },
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=4",
+    "node": ">=18.0.0",
+    "npm": "please-use-yarn",
```
Comment (kpanot) [inline packages/@ama-sdk/core/package.json:126]: note mandatory

File: packages/@ama-sdk/core/package.json [hunk 3]
```diff
@@ -122,8 +122,7 @@
     "zone.js": "~0.13.1"
   },
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=4",
+    "node": ">=18.0.0",
     "yarn": "<4.0.0"
```
Comment (kpanot) [inline packages/@ama-sdk/core/package.json:126]: can be removed as well

File: packages/@ama-sdk/generator-sdk/package.json [hunk 4]
```diff
@@ -85,8 +85,7 @@
     "yeoman-test": "^6.0.0"
   },
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=4",
+    "node": ">=18.0.0",
     "yarn": "<4.0.0"
```
Comment (kpanot) [inline packages/@ama-sdk/generator-sdk/package.json:89]: can be removed

File: packages/@ama-sdk/generator-sdk/src/generators/shell/templates/base/package.json [hunk 5]
```diff
@@ -119,8 +119,7 @@
     "isomorphic-fetch": "~3.0.0"
   },
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=6.0.0",
+    "node": ">=18.0.0",
     "yarn": ">=3.1.0"
```
Comment (kpanot) [inline packages/@ama-sdk/generator-sdk/src/generators/shell/templates/base/package.json:123]: Can be removed

File: packages/@ama-sdk/schematics/package.json [hunk 6]
```diff
@@ -99,8 +99,7 @@
   },
   "openApiSupportedVersion": "~6.6.0",
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=4",
+    "node": ">=18.0.0",
     "yarn": "<4.0.0"
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/package.json:103]: can be removed

File: packages/@ama-sdk/swagger-builder/package.json [hunk 7]
```diff
@@ -58,8 +58,7 @@
     "typescript": "~5.1.6"
   },
   "engines": {
-    "node": ">=16.0.0",
-    "npm": ">=4",
+    "node": ">=18.0.0",
     "yarn": "<4.0.0"
```
Comment (kpanot) [inline packages/@ama-sdk/swagger-builder/package.json:62]: can be removed

**Review by fpaul-1A — APPROVED** [review]

File: tools/github-actions/get-npm-tag/action.yml [hunk 8]
```diff
@@ -22,5 +22,5 @@ outputs:
     description: 'NPM Tag to apply'
     value: ${{ steps.get-npm-tag.outputs.tag }}
 runs:
-  using: 'node16'
+  using: 'node20'
```
Comment (kpanot) [inline tools/github-actions/get-npm-tag/action.yml:25]: why not `node18` ?

**Review by kpanot — APPROVED** [review]

File: tools/github-actions/get-npm-tag/action.yml (hunk 8 above)
Comment (vscaiceanu-1a) [reply tools/github-actions/get-npm-tag/action.yml:25]: because`node18` doesn't exist https://github.com/orgs/community/discussions/53217

**Pull request merged**